src/java.desktop/share/classes/com/sun/java/swing/plaf/gtk/GTKPainter.java

```diff
@@ -610,9 +610,9 @@ public void paintPopupMenuBackground(SynthContext context, Graphics g,
                 x + insets.left, y + insets.top, w - insets.left - insets.right,
                 h - insets.top - insets.bottom);
             BufferedImage img = ENGINE.finishPainting();
-            if(!isHW) {
+            if(!isHW && img != null) {
                 int border = img.getRGB(0, h / 2);
-                if (img != null && border == img.getRGB(w / 2, h / 2)) {
+                if (border == img.getRGB(w / 2, h / 2)) {
                     // fix no menu borders in Adwaita theme
```

Comment:
since it is needed for Yaru also, then I guess it's better to remove specific mention of Adwaita

Reply:
Ok... updated.